### PR TITLE
Fitsheader: Add new `--sort` argument

### PR DESCRIPTION
**Pull request by krachyon**
### Description
Continuation of https://github.com/astropy/astropy/pull/13011, with the "extra argument" solution. 
Imo it only really makes sense in 'fitsort' mode as the other two create multiple lines of output per file. I could add it to the others for consistency.

It isn't the most elegant as there seems to be no way to express "this argument needs this other argument to be present" in argparse. Another thing i noticed though is that the `-t` argument has similar issues than adding `nargs='?'` for `--fitsort`.

Marked WIP so far, to address added todo comments and questions here.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/io/fits/scripts/fitsheader.py [hunk 1]
```diff
@@ -451,6 +458,20 @@ def main(args=None):
...
+    if args.sort:
+        args.sort = [key.replace('.', ' ') for key in args.sort]
+        if not args.fitsort:
+            # TODO Only really makes sense for fitsort format. Maybe for standard
+            #  but then it would chug for a while and not start printing
+            #  immediately if you pass many files.
+            #  Maybe this should be an error, but that seems hard
+            #  to do natively with argparse, i.e. exception looks different
+            log.warning('Sorting is only supported in conjunction with -f/--fitsort. '
+                        'Sorting keywords are ignored.')
```
Comment (saimn) [inline astropy/io/fits/scripts/fitsheader.py:473]: I think it's fine to support sorting only with fitsort. And yes it's probably better to raise an exception, or just use `sys.exit` to terminate the program (not sure what's the issue with an exception, but I usually use `sys.exit` in that case).

File: docs/changes/io.fits/13106.api.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Add ``-s/--sort`` argument to ``fitsheader`` to sort the fitsort-mode output::
+``fitsheader -f -s/--sort [SORT_KEYWORD]``; Removed docs/code for passing
+``SORT_KEYWORD`` as an optional argument to ``-f``
```
Comment (saimn) [inline docs/changes/io.fits/13106.api.rst:3]:
```suggestion
Add ``-s/--sort`` argument to ``fitsheader`` to sort the fitsort-mode output.
```
I think the short version is sufficient (sort wasn't working before).

File: astropy/io/fits/scripts/fitsheader.py [hunk 3]
```diff
@@ -451,6 +458,20 @@ def main(args=None):
     if args.table is None:
         args.table = 'ascii.fixed_width'
 
+    # TODO: This is imho a little nicer than the custom action approach requiring
+    #  the store keyword.
+    #  Approach should be unified to whatever method is chosen
```
Comment (saimn) [inline astropy/io/fits/scripts/fitsheader.py:463]: Yes agreed, could you do the same for keywords ? Thanks.

**Review by saimn — COMMENTED** [review]
Thanks for the PR, @krachyon. It just needs a few minor changes, would be great to have this by the end of the week so it can be included in 5.1.

File: astropy/io/fits/scripts/fitsheader.py (hunk 1 above)
Comment (krachyon) [reply astropy/io/fits/scripts/fitsheader.py]: I was trying to find a way to get `argparse` to report the error, like it does for other invalid arguments.
```
usage: fitsheader [-h] [--version] [-e HDU] [-k KEYWORD] [-t [FORMAT]] [-f] [-c] filename [filename ...]
fitsheader: error: the following arguments are required: filename
```
An exception prints the whole stack-trace so it looks quite different to the user. Adopted `sys.exit` method

File: astropy/io/fits/scripts/fitsheader.py (hunk 3 above)
Comment (krachyon) [reply astropy/io/fits/scripts/fitsheader.py]: Done

**Pull request merged**